src/main/java/com/iexec/worker/compute/ComputeStage.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.worker.compute;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class ComputeStage {
```

Comment:
I used it for that https://github.com/iExecBlockchainComputing/iexec-worker/blob/391ce1c4ae4e3f1cc837b8ac6223500ad8bb7d1b/src/main/java/com/iexec/worker/compute/ComputeController.java#L66-L66, do you have an option for it?

Reply:
Maybe something like:
```
public enum ComputeStage {
  PRE("pre"),
  POST("post");

  private final String stageName;

  ComputeStage(String stageName) {
    this.stageName = stageName;
   }

  public String getStageName() { return this.stageName; }
}
```

And:

```
"/compute/" + ComputeStage.POST.getStageName() + "/{chainTaskId}/computed"
```
?

Reply:
Yes, have you tried .. :) ?

Reply:
2 options are possible, the one suggested by maxence or by directly defining Enum values in lowercase
`Enum ComputeStage { pre, post }`
By default `Compute.pre.toString()` is `"pre"`.
All `String computeStage` usages can be replaced with `ComputeStageEnum computeStage`. (I used ComputeStageEnum suffix for my test to avoid overriding ComputeStage class)
As it offers type safety, the `ComputeStage#isValid` method becomes irrelevant and no request can be accepted if `computeStage` is not a valid value from the ComputeStageEnum.
You can try it for yourself to see how this works.

Reply:
Totally wanted from the start yes! But I ran into that issue with an enum:
https://github.com/iExecBlockchainComputing/iexec-worker/blob/391ce1c4ae4e3f1cc837b8ac6223500ad8bb7d1b/src/main/java/com/iexec/worker/compute/ComputeController.java#L66-L66
`Attribute value must be constant`.

Here is the solution for it:
https://github.com/iExecBlockchainComputing/iexec-worker/blob/026b318899cc7ade001af56ca05884eda7da4c41/src/main/java/com/iexec/worker/compute/ComputeStage.ja
```
public class ComputeStage {

    PRE(ComputeStage.PRE_VALUE),
    POST(ComputeStage.POST_VALUE);

    public static final String PRE_VALUE = "pre";
    public static final String POST_VALUE = "post";

    private final String value;

    ComputeStage(String value) {
        this.value = value;
    }
}
```
```
    @PostMapping(path = {
            "/compute/" + ComputeStage.POST_VALUE + "/{chainTaskId}/computed"
    })
```
Fixed in https://github.com/iExecBlockchainComputing/iexec-worker/pull/447/commits/026b318899cc7ade001af56ca05884eda7da4c41. :)